src/Core/SqlManager/Form/SqlRequestFormDataProvider.php

```diff
@@ -0,0 +1,138 @@
+use PrestaShop\PrestaShop\Core\Domain\SqlManagement\Query\GetSqlRequestForEditing;
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObjectFormDataProviderInterface;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class SqlRequestFormDataProvider.
+ */
+final class SqlRequestFormDataProvider implements IdentifiableObjectFormDataProviderInterface
+{
+    /**
+     * @var Request
+     */
+    private $request;
+
+    /**
+     * @var CommandBusInterface
+     */
+    private $queryBus;
+
+    /**
+     * @var HookDispatcherInterface
+     */
+    private $hookDispatcher;
+
+    /**
+     * @param CommandBusInterface $queryBus
+     * @param HookDispatcherInterface $hookDispatcher
+     * @param Request $request
+     */
+    public function __construct(
+        CommandBusInterface $queryBus,
+        HookDispatcherInterface $hookDispatcher,
+        Request $request = null
+    ) {
+        $this->request = $request;
+        $this->queryBus = $queryBus;
+        $this->hookDispatcher = $hookDispatcher;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData($sqlRequestId = null)
+    {
+        $sqlRequestFormData = $this->getSqlRequestFormData($sqlRequestId);
+
+        $this->hookDispatcher->dispatchWithParameters('actionSqlRequestFormDataModifier', [
+            'id' => $sqlRequestId,
+            'data' => &$sqlRequestFormData,
+        ]);
+    }
+
+    /**
+     * @param int|null $sqlRequestId
+     *
+     * @return array
+     */
+    private function getSqlRequestFormData($sqlRequestId)
+    {
+        if (null === $sqlRequestId) {
+            return $this->getEmptySqlRequestFormData();
+        }
+
+        return $this->getPersistedSqlRequestFormData($sqlRequestId);
+    }
+
+    /**
+     * Get data for empty form.
+     *
+     * @return array
+     */
+    private function getEmptySqlRequestFormData()
+    {
+        if ($this->request && $this->request->request->has('sql')) {
+            return [
+                'sql' => $this->request->request->get('sql'),
+                'name' => $this->request->request->get('name'),
+            ];
+        }
+
+        return [];
+    }
+
+    /**
+     * @param int $sqlRequestId
+     *
+     * @return array
+     */
+    private function getPersistedSqlRequestFormData($sqlRequestId)
+    {
+        try {
+            $getRequestSqlForEditingQuery = new GetSqlRequestForEditing($sqlRequestId);
+
+            /** @var EditableSqlRequest $editableRequestSql */
+            $editableRequestSql = $this->queryBus->handle($getRequestSqlForEditingQuery);
+
+            return [
+                'id' => $editableRequestSql->getSqlRequestId()->getValue(),
+                'name' => $editableRequestSql->getName(),
+                'sql' => $editableRequestSql->getSql(),
+            ];
+        } catch (SqlRequestException $e) {
+            return [];
```

Comment:
Is that the expected behavior ? If the SQL request fail, we show the form with no data ?